src/lib/fast_check.py

```diff
@@ -0,0 +1,123 @@
+
+
+def find_service_name(service):
+    return service_name_pattern.match(service).group(2) if service_name_pattern.match(service) else None
+
+
+class FastCheck:
+
+    def __init__(self, session: ClientSession, token: str, logging_context: LoggingContext):
+        self.session = session
+        self.logging_context = logging_context
+        self.token = token
+
+    async def list_services(self, project_id: str, timeout: Optional[int] = 2):
+        try:
+            response = await self.session.get(
+                urljoin(GCP_SERVICE_USAGE_URL, f'{project_id}/services'),
+                headers={
+                    "Authorization": f'Bearer {self.token}'
+                },
+                params={
+                    "filter": "state:ENABLED"
+                },
+                timeout=timeout)
+            if response.status != 200:
+                self.logging_context.log(f'Http error: {response.status}, url: {response.url}, reason: {response.reason}')
+                return None
+
+            return await response.json()
+        except Exception as e:
+            self.logging_context.log(f'Unable to get project: {project_id} services list. Error details: {e}')
+            return None
+
+    async def get_dynatrace_token_metadata(self, dynatrace_url: str, dynatrace_api_key: str, timeout: Optional[int] = 2):
+        try:
+            response = await self.session.post(
+                url=urljoin(dynatrace_url, "/api/v1/tokens/lookup"),
+                headers={
+                    "Authorization": f"Api-Token {dynatrace_api_key}",
+                    "Content-Type": "application/json; charset=utf-8"
+                },
+                json={
+                    "token": dynatrace_api_key
+                },
+                timeout=timeout)
+            if response.status != 200:
+                self.logging_context.log(f'Unable to get Dynatrace token metadata: {response.status}, url: {response.url}, reason: {response.reason}')
+                return None
+
+            return await response.json()
+        except Exception as e:
+            self.logging_context.log(f'Unable to get Dynatrace token metadata. Error details: {e}')
+            return None
+
+    async def _check_services(self, project_id):
+        list_services_result = await self.list_services(project_id)
+        service_names = [find_service_name(service['name']) for service in list_services_result['services']] if list_services_result else []
+        if not all(name in service_names for name in required_services):
+            self.logging_context.log(f'Cannot monitor project: \'{project_id}\'. '
+                                     f'Enable required services: {required_services}')
+            return None
+        return service_names
+
+    async def _check_dynatrace(self, project_id):
+        try:
+            dynatrace_url = await fetch_secret(self.session, project_id, self.token, "DYNATRACE_URL")
```

Comment:
You should use `fetch_dynatrace_url` and `fetch_dynatrace_api_key` functions from `credentials.py` instead. They fetch the relevant secret name from environment variables.